### PR TITLE
:sparkles: [2FA] Can activate 2fa during user creation

### PR DESCRIPTION
**Pull request by NilsRec**
[Ticket](link_to_trello_issue)

### CODE REVIEW

**Review by tcheymol — APPROVED** [review]

**Review by AntoineAresu — APPROVED** [review]

**Review by tcheymol — APPROVED** [review]

File: templates/v2/user_creation/components/_password_card.html.twig [hunk 1]
```diff
@@ -2,12 +2,19 @@
...
+        {% for field in form %}
+            {#          The 'plainPassword' field name is only used when creating a professionnal #}
+            {% if field.vars.name == 'password' %}
+                {{ include('v2/user/strong_password/_strong_password_row.html.twig', {
+                    'field': form.password,
+                    'isBeneficiaire': true,
+                    'hideAndShow': false,
+                    'generatePassword': true,
+                }) }}            {% else %}
+                {{ form_row(field) }}
```
Comment (AntoineAresu) [inline templates/v2/user_creation/components/_password_card.html.twig]: little indent mistake

File: config/packages/security.yaml [hunk 2]
```diff
@@ -75,6 +75,7 @@ security:
         - { path: ^/admin/app/user, roles: ROLE_SUPER_ADMIN }
         - { path: ^/admin, roles: ROLE_ADMIN }
         # This ensures that the form can only be accessed when two-factor authentication is in progress.
+        - { path: ^/resend-auth-code, role: IS_AUTHENTICATED_2FA_IN_PROGRESS }
         - { path: ^/2fa, role: IS_AUTHENTICATED_2FA_IN_PROGRESS }
```
Comment (AntoineAresu) [inline config/packages/security.yaml:79]: `- { path: ^/(resend-auth-code|2fa), roles: IS_AUTHENTICATED_2FA_IN_PROGRESS }`

File: src/Controller/SecurityController.php [hunk 3]
```diff
@@ -61,4 +66,27 @@ public function loginLink()
     {
         throw new \LogicException('This code should never be reached');
     }
+
+    #[Route('/resend-auth-code', name: 'resend_auth_code')]
```
Comment (AntoineAresu) [inline src/Controller/SecurityController.php:70]: [Route('/resend-auth-code', name: 'resend_auth_code', methods: ['GET'])]

File: src/Entity/User.php [hunk 4]
```diff
@@ -1412,4 +1413,30 @@ public function setMfaMethod(string $mfaMethod): User
...
+
+        return $this;
+    }
+
+    public function increaseMfaRetryCount(): User
+    {
+        ++$this->mfaRetryCount;
+
+        return $this;
+    }
```
Comment (AntoineAresu) [inline src/Entity/User.php:1441]: `return static`

Comment (AntoineAresu) [reply src/Entity/User.php:1441]: Same for previous mfa methods

File: src/Controller/SecurityController.php [hunk 5]
```diff
@@ -61,4 +66,27 @@ public function loginLink()
...
+
+        if (self::MFA_MAX_RETRIES === $user->getMfaRetryCount()) {
+            $this->addFlash('danger', 'mfa_maximum_retries_reach');
+
+            return $this->redirectToRoute('2fa_login');
+        }
+
+        User::MFA_METHOD_EMAIL === $user->getMfaMethod() ? $emailCodeGenerator->generateAndSend($user) : $smsCodeGenerator->generateAndSend($user);
+        $user->increaseMfaRetryCount();
+        $em->flush();
```
Comment (AntoineAresu) [inline src/Controller/SecurityController.php:88]: What do you think about a separated service ? Not sure of the naming but
`$service->sendCode($user);` could be nice

File: translations/messages.fr.yml [hunk 6]
```diff
@@ -631,4 +631,7 @@ mfa_method: "Recevoir le code par :"
...
 no_phone_number: Vous avez choisi de recevoir un code de connexion par SMS, mais vous n'avez pas renseigné de numéro de téléphone.
-mfa_description: L'authentification à deux facteurs vous permet de recevoir un code par email ou sms à chaque connexion pour valider votre identité et sécuriser votre compte. Si vous souhaitez activer cette authentification à deux facteurs, veuillez cocher la case ci-dessous et choisir la méthode par laquelle vous souhaitez recevoir votre code de connexion.
\ No newline at end of file
+mfa_description: L'authentification à deux facteurs vous permet de recevoir un code par email ou sms à chaque connexion pour valider votre identité et sécuriser votre compte. Si vous souhaitez activer cette authentification à deux facteurs, veuillez cocher la case ci-dessous et choisir la méthode par laquelle vous souhaitez recevoir votre code de connexion.
+mfa_maximum_retries_reach: Vous avez atteint le nombre maximum de renvois autorisés. Si vous ne parvenez pas à vous connecter, veuillez contacter le support.
+mfa_code_not_received: Vous n'avez pas reçu le code ?
+resend_mfa_code: Cliquez ici pour le renvoyer
```
Comment (AntoineAresu) [inline translations/messages.fr.yml:637]: newline

File: translations/security/security.fr.yml [hunk 7]
```diff
@@ -0,0 +1 @@
+code_invalid: Le code est incorrect
```
Comment (AntoineAresu) [inline translations/security/security.fr.yml:1]: newline

File: src/FormV2/MfaType.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                'expanded' => true,
+                'multiple' => false,
+            ]);
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => User::class,
+            'validation_groups' => [$this->getUser()?->isBeneficiaire() ? 'beneficiaire' : 'membre'],
```
Comment (tcheymol) [inline src/FormV2/MfaType.php:43]: On pourrait même avoir une méthode $user->getValidationGroup()

**Review by tcheymol — APPROVED** [review]

**Review by AntoineAresu — APPROVED** [review]

**Pull request merged**